src/js/components/stock-movement/inbound/StockMovementInboundTable.jsx

```diff
@@ -0,0 +1,399 @@
+  const [pages, setPages] = useState(-1);
+  // Stored searching params for export case
+  const [currentParams, setCurrentParams] = useState({});
+  const [totalData, setTotalData] = useState(0);
+
+  const tableRef = useRef(null);
+
+  const fireFetchData = () => tableRef.current.fireFetchData();
+
+  // If filterParams change, refetch the data with applied filters
+  useEffect(() => {
+    fireFetchData();
+    if (!isShipmentStatusesFetched) fetchStatuses();
+  }, [filterParams]);
+
+  const exportStockMovements = () => {
+    apiClient.get('/openboxes/api/stockMovements', {
+      params: {
+        ...currentParams,
+        format: 'csv',
+      },
+      paramsSerializer: params => queryString.stringify(params),
+    })
+      .then((res) => {
+        const filename = res.headers['content-disposition'].split('filename="')[1].split('.')[0];
+        fileDownload(res.data, filename, 'text/csv');
+      });
+  };
+
+  const exportAllIncomingItems = () => {
+    apiClient.get('/openboxes/api/stockMovements/shippedItems', {
+      params: {
+        ...currentParams,
+        format: 'csv',
+      },
+      paramsSerializer: params => queryString.stringify(params),
+    })
+      .then((res) => {
+        const filename = res.headers['content-disposition'].split('filename="')[1].split('.')[0];
+        fileDownload(res.data, filename, 'text/csv');
+      });
+  };
+
+
+  const getStatusTooltip = status => translate(
+    `react.stockMovement.status.${status.toLowerCase()}.description.label`,
+    status.toLowerCase(),
+  );
+
+
+  const onFetchHandler = (state) => {
+    const offset = state.page > 0 ? (state.page) * state.pageSize : 0;
+    const sortingParams = state.sorted.length > 0 ?
+      {
+        sort: state.sorted[0].id,
+        order: state.sorted[0].desc ? 'desc' : 'asc',
+      } : undefined;
+
+    const params = _.omitBy({
+      ...filterParams,
+      offset: `${offset}`,
+      max: `${state.pageSize}`,
+      direction: 'INBOUND',
+      receiptStatusCode: filterParams.receiptStatusCode &&
+        filterParams.receiptStatusCode?.map(({ id }) => id),
+      origin: filterParams?.origin?.id,
+      destination: filterParams?.destination?.id,
+      requestedBy: filterParams.requestedBy?.id,
+      createdBy: filterParams.createdBy?.id,
+      updatedBy: filterParams.updatedBy?.id,
+      ...sortingParams,
+    }, (value) => {
+      if (typeof value === 'object' && _.isEmpty(value)) return true;
+      return !value;
+    });
+
+    // Fetch data
+    setLoading(true);
+    apiClient.get('/openboxes/api/stockMovements', {
+      paramsSerializer: parameters => queryString.stringify(parameters),
+      params,
+    })
+      .then((res) => {
+        setLoading(false);
+        setPages(Math.ceil(res.data.totalCount / state.pageSize));
+        setTotalData(res.data.totalCount);
+        setTableData(res.data.data);
+        // Store currently used params for export case
+        setCurrentParams(params);
```

Comment:
as discussed yesterday, maybe do one object-state, so you don't have to make multiple `setStates` where each of them re-renders component?